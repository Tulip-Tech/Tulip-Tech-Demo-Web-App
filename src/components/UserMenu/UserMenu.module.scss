@use 'src/styles/variables';
@use 'src/styles/theme';

.panel {
  width: 100%;
  height: 100%;
  padding: variables.$base-spacing 0;
  font-weight: normal;
  font-size: 16px;
  font-style: normal;
  line-height: 18px;
  letter-spacing: 0.15px;
  background: theme.$modal-bg;
  box-shadow: theme.$panel-box-shadow;
}

.menuItems {
  margin: 0;
  padding: 0;
  list-style-type: none;

  > li > a {
    padding: 0 18px;
    font-size: 16px;
  }
}

.button {
  margin-bottom: variables.$base-spacing;
}

.divider {
  vertical-align: baseline;
  background: transparent;
  border: 0;
  border-top: 1px solid rgba(variables.$white, 0.12);

  &.inPopover {
    margin: variables.$base-spacing;
    border-top: 1px solid rgba(255, 255, 255, 0.32);
  }
}
