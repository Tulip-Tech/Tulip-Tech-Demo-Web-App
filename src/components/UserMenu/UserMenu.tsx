import React, { useCallback } from 'react';
import { useNavigate } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import classNames from 'classnames';

import styles from './UserMenu.module.scss';

import AccountCircle from '#src/icons/AccountCircle';
import Favorite from '#src/icons/Favorite';
import BalanceWallet from '#src/icons/BalanceWallet';
import Exit from '#src/icons/Exit';
import MenuButton from '#components/MenuButton/MenuButton';
import { logout } from '#src/stores/AccountController';

type Props = {
  inPopover?: boolean;
  showPaymentsItem: boolean;
  onClick?: () => void;
};

const UserMenu = ({ showPaymentsItem, inPopover = false, onClick }: Props) => {
  const { t } = useTranslation('user');
  const navigate = useNavigate();

  const onLogout = useCallback(async () => {
    if (onClick) {
      onClick();
    }

    await logout();
    navigate('/', { replace: true });
  }, [onClick, navigate]);

  const menuItems = (
    <ul className={styles.menuItems}>
      <li>
        <MenuButton small={inPopover} onClick={onClick} to="/u/my-account" label={t('nav.account')} startIcon={<AccountCircle />} />
      </li>
      <li>
        <MenuButton small={inPopover} onClick={onClick} to="/u/favorites" label={t('nav.favorites')} startIcon={<Favorite />} />
      </li>
      {showPaymentsItem && (
        <li>
          <MenuButton small={inPopover} onClick={onClick} to="/u/payments" label={t('nav.payments')} startIcon={<BalanceWallet />} />
        </li>
      )}
      <hr className={classNames(styles.divider, { [styles.inPopover]: inPopover })} />
      <li>
        <MenuButton small={inPopover} onClick={onLogout} label={t('nav.logout')} startIcon={<Exit />} />
      </li>
    </ul>
  );

  if (inPopover) {
    return <nav className={styles.panel}>{menuItems}</nav>;
  }

  return menuItems;
};

export default UserMenu;
