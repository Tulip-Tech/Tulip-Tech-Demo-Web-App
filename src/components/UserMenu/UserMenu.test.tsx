import React from 'react';

import UserMenu from './UserMenu';

import { renderWithRouter } from '#test/testUtils';

describe('<UserMenu>', () => {
  test('renders and matches snapshot', () => {
    const { container } = renderWithRouter(<UserMenu showPaymentsItem={true} />);

    expect(container).toMatchSnapshot();
  });
});
