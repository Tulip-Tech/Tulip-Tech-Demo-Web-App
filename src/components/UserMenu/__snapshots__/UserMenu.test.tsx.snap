// Vitest Snapshot v1

exports[`<UserMenu> > renders and matches snapshot 1`] = `
<div>
  <ul
    class="_menuItems_457528"
  >
    <li>
      <a
        aria-label="nav.account"
        class="_menuButton_91706b"
        href="/u/my-account"
        tabindex="0"
      >
        <div
          class="_startIcon_91706b"
        >
          <svg
            aria-hidden="true"
            class="_icon_1e9999"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M12,19.2C9.5,19.2 7.29,17.92 6,16C6.03,14 10,12.9 12,12.9C14,12.9 17.97,14 18,16C16.71,17.92 14.5,19.2 12,19.2M12,5A3,3 0 0,1 15,8A3,3 0 0,1 12,11A3,3 0 0,1 9,8A3,3 0 0,1 12,5M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12C22,6.47 17.5,2 12,2Z"
              />
            </g>
          </svg>
        </div>
        <span>
          nav.account
        </span>
      </a>
    </li>
    <li>
      <a
        aria-label="nav.favorites"
        class="_menuButton_91706b"
        href="/u/favorites"
        tabindex="0"
      >
        <div
          class="_startIcon_91706b"
        >
          <svg
            aria-hidden="true"
            class="_icon_1e9999"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
              />
            </g>
          </svg>
        </div>
        <span>
          nav.favorites
        </span>
      </a>
    </li>
    <li>
      <a
        aria-label="nav.payments"
        class="_menuButton_91706b"
        href="/u/payments"
        tabindex="0"
      >
        <div
          class="_startIcon_91706b"
        >
          <svg
            aria-hidden="true"
            class="_icon_1e9999"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M5,3C3.89,3 3,3.9 3,5V19A2,2 0 0,0 5,21H19A2,2 0 0,0 21,19V16.72C21.59,16.37 22,15.74 22,15V9C22,8.26 21.59,7.63 21,7.28V5A2,2 0 0,0 19,3H5M5,5H19V7H13A2,2 0 0,0 11,9V15A2,2 0 0,0 13,17H19V19H5V5M13,9H20V15H13V9M16,10.5A1.5,1.5 0 0,0 14.5,12A1.5,1.5 0 0,0 16,13.5A1.5,1.5 0 0,0 17.5,12A1.5,1.5 0 0,0 16,10.5Z"
              />
            </g>
          </svg>
        </div>
        <span>
          nav.payments
        </span>
      </a>
    </li>
    <hr
      class="_divider_457528"
    />
    <li>
      <div
        aria-label="nav.logout"
        class="_menuButton_91706b"
        tabindex="0"
      >
        <div
          class="_startIcon_91706b"
        >
          <svg
            aria-hidden="true"
            class="_icon_1e9999"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M19,3H5C3.89,3 3,3.89 3,5V9H5V5H19V19H5V15H3V19A2,2 0 0,0 5,21H19A2,2 0 0,0 21,19V5C21,3.89 20.1,3 19,3M10.08,15.58L11.5,17L16.5,12L11.5,7L10.08,8.41L12.67,11H3V13H12.67L10.08,15.58Z"
              />
            </g>
          </svg>
        </div>
        <span>
          nav.logout
        </span>
      </div>
    </li>
  </ul>
</div>
`;
